### ISSUE TITLE
deleteByPattern returns zero

### ISSUE THREAD

**Issue by HaroldFromRovia** [redisson/redisson#4711]
**Expected behavior**

deleteByPattern returns actual number of deleted rows  

**Actual behavior**

deleteByPattern returns 0

**Steps to reproduce or test case**

I've got structure like this
![image](https://user-images.githubusercontent.com/28599564/205061703-567e0c98-2db4-4eb6-8dc4-1b2e54ff7985.png)
Code below always returns 0
```
    public void deleteAll() {
        long removedCount = client.getKeys()
                .deleteByPattern("/key/*");
        log.trace("Removed {}", removedCount);
    }
```

Also I've noticed that if we have a large amount of data and try to put something iside that map, that data will be removed after a short time of duration (delete executes on server after put). Reproduces only on remote server, locally works fine.

**Redis version**

6.2.6 (also reproduces on 6.2.7 and 5.0.7)

**Redisson version**

3.16.8

**Redisson configuration**
```
    @Bean
    public RedissonClient client() {
        Config config = new Config();
        config.useSingleServer()
                .setAddress(properties.getAddress());
                
        return Redisson.create(config);
    }
```

**Comment by HaroldFromRovia** [comment]
On 3.18.0 put after delete works correctly, but method still returns 0

**Comment by HaroldFromRovia** [comment]
Fixed it by upgrading version to 3.18.0
